pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -458,14 +458,94 @@ def ctrl_decomp_bisect(
 
 
 def decompose_mcx(control_wires, target_wire, work_wires):
-    """Decomposes the multi-controlled PauliX gate"""
-
+    """Decomposes the multi-controlled PauliX gate using decompositions from
+    `Barenco et al. (1995) <https://arxiv.org/abs/quant-ph/9503016>`_"""
     num_work_wires_needed = len(control_wires) - 2
 
+    if len(control_wires) == 1:
+        return [qml.CNOT(wires=control_wires + Wires(target_wire))]
+    if len(control_wires) == 2:
+        return qml.Toffoli.compute_decomposition(wires=control_wires + Wires(target_wire))
+
     if len(work_wires) >= num_work_wires_needed:
+        # Lemma 7.2
         return _decompose_mcx_with_many_workers(control_wires, target_wire, work_wires)
+    if len(work_wires) >= 1:
+        # Lemma 7.3
+        return _decompose_mcx_with_one_worker(control_wires, target_wire, work_wires[0])
+
+    # Lemma 7.5
+    with qml.QueuingManager.stop_recording():
+        op = qml.X(target_wire)
+    return _decompose_multicontrolled_unitary(op, control_wires)
+
+
+def _decompose_multicontrolled_unitary(op, control_wires):
+    """Decomposes general multi controlled unitary with no work wires
+    Follows approach from Lemma 7.5 combined with 7.3 and 7.2 of
+    https://arxiv.org/abs/quant-ph/9503016.
+
+    We are assuming this decomposition is used only in the general cases
+    """
+    # pylint: disable=import-outside-toplevel
+    from pennylane.ops.op_math.controlled import _is_single_qubit_special_unitary
+
+    if not op.has_matrix or len(op.wires) != 1:
+        raise ValueError(
+            "The target operation must be a single-qubit operation with a matrix representation"
+        )
+
+    target_wire = op.wires
+    if len(control_wires) == 0:
+        return [op]
+    if len(control_wires) == 1:
+        return ctrl_decomp_zyz(op, control_wires)
+    if _is_single_qubit_special_unitary(op):
+        return ctrl_decomp_bisect(op, control_wires)
```

Comment:
Note that thethe `bisect` algorithm isn't currently backprop compatible due to calculating the eigenvalues and eigenvectors. So while this is fine for a `X` operation, it may cause issues with a general operation that may be trainable.

Reply:
Oh I see. So I guess one possibility to make `_decompose_multicontrolled_unitary` backprop compatible would be to use `_decompose_recursive` whenever `op` is trainable, right? (regardless `op` is in SU(2) or not). That is of course assuming we can make sure `_decompose_recursive ` is differentiable. I am honestly not acquainted with how to make sure of that.